heimlig/src/crypto/ecdsa.rs

```diff
@@ -1,60 +1,630 @@
+        $private_key_size:ident,
+        $private_key_size_str:expr,
+        $public_key_size:ident,
+        $public_key_size_str:expr,
+        $doc:expr
+    ) => {
+        #[doc=$doc]
+        /// signature size in bytes.
+        pub const $signature_size: usize = SignatureSize::<$curve>::USIZE;
+
+        #[doc=$doc]
+        /// digest size in bytes.
+        pub const $digest_size: usize = DigestSize::<$curve>::USIZE;
+
+        #[doc=$doc]
+        /// private key size in bytes.
+        pub const $private_key_size: usize = PrivateKeySize::<$curve>::USIZE;
+
+        #[doc=$doc]
+        /// public key size in bytes.
+        pub const $public_key_size: usize = PublicKeySize::<$curve>::USIZE;
+
+        #[doc=$doc]
+        /// signing the message function.
+        ///
+        ///  # Arguments
+        ///
+        /// * `private_key`: A slice containing private key bytes.
+        ///   The private key has to be `
+        #[doc=$private_key_size_str]
+        /// ` bytes long.
+        /// * `message`: A slice containing the message to sign bytes.
+        /// * `signature`: A mutable slice where the computed signature will be stored.
+        ///   The signature slice length has to be `
+        #[doc=$signature_size_str]
+        /// ` bytes long.
+        ///
+        /// # Errors
+        ///
+        /// The function returns an error if:
+        /// * `InvalidSignatureSize`: The length of the `signature` is not `
+        #[doc=$signature_size_str]
+        /// ` bytes.
+        /// * `InvalidPrivateKey`: The length of the `private_key` is not `
+        #[doc=$private_key_size_str]
+        /// ` bytes.
+        pub fn $sign(
+            private_key: &[u8],
+            message: &[u8],
+            signature: &mut [u8],
+        ) -> Result<(), Error> {
+            sign::<$curve>(private_key, message, signature)
+        }
+
+        #[doc=$doc]
+        /// signing prehashed message function.
+        ///
+        ///  # Arguments
+        ///
+        /// * `private_key`: A slice containing private key bytes.
+        ///   The private key has to be `
+        #[doc=$private_key_size_str]
+        /// ` bytes long.
+        /// * `digest`: A slice containing the digest to sign bytes.
+        ///    The digest has to be `
+        #[doc=$digest_size_str]
+        /// ` bytes long.
+        /// * `signature`: A mutable slice where the computed signature will be stored.
+        ///   The signature slice length has to be `
+        #[doc=$signature_size_str]
+        /// ` bytes long.
+        ///
+        /// # Errors
+        ///
+        /// The function returns an error if:
+        /// * `InvalidSignatureSize`: The length of the `signature` is not `
+        #[doc=$signature_size_str]
+        /// ` bytes.
+        /// * `InvalidDigest`: The length of the `digest` is not `
+        #[doc=$digest_size_str]
+        /// ` bytes.
+        /// * `InvalidPrivateKey`: The length of the `private_key` is not `
```

Comment:
Should we rename our size errors to `InvalidDigestSize` and `InvalidPrivateKey` to stay consistent with `InvalidSignatureSize`?

Reply:
Let me create it as a separate PR to fix it also for `ed25519`.